R/gen_population.R

```diff
@@ -6,66 +6,128 @@
+#'   have the length equal to groups times strata (or half thereof).
+#'   These defaults are based of MSF general emergency intervention standard
+#'   values.
+#' @param counts A numeric vector specifying the counts for each group.
+#'   The default is NULL - as most often proportions above will be used.
+#'   If is not NULL then total_pop and proportions will be ignored.
+#'   Make sure the length of this vector is equal to groups times strata (or if it
+#'   is half then it will repeat for each strata).
+#'   For reference, the MSF general emergency intervention standard values
+#'   are c(7945, 13391, 13861, 8138, 6665) based on above
+#'   groups for a 100,000 person population.
 #' @param tibble Return data as a tidyverse tibble (default is TRUE)
-#' @importFrom dplyr bind_cols
+#' @importFrom glue glue
+#' @importFrom tibble as_tibble
 #' @export
+#'
+#' @examples
+#' # get population counts based on proportion, unstratified
+#' gen_population(groups = c(1, 2, 3, 4), strata = NULL, proportions = c(0.3, 0.2, 0.4, 0.1))
+#'
+#' # get population counts based on proportion, stratified
+#' gen_population(groups = c(1, 2, 3, 4), strata = c("a", "b"), proportions = c(0.3, 0.2, 0.4, 0.1))
+#'
+#' # get population counts based on counts, unstratified
+#' gen_population(groups = c(1, 2, 3, 4), strata = NULL, counts = c(20, 10, 30, 40))
+#'
+#' # get population counts based on counts, stratified
+#' gen_population(groups = c(1, 2, 3, 4), strata = c("a", "b"), counts = c(20, 10, 30, 40))
+#'
+#' # get population counts based on counts, stratified - type out counts for each group and strata
+#' gen_population(groups = c(1, 2, 3, 4), strata = c("a", "b"), counts = c(20, 10, 30, 40, 40, 30, 20, 20))
 
 
 gen_population <- function(total_pop = 1000,
-                           groups = c("0-4","5-14","15-24","25-34","35-44","45+"),
+                           groups = c("0-4","5-14","15-29","30-44","45+"),
                            strata = c("Male", "Female"),
-                           proportions = c(0.182, 0.278, 0.26, 0.11, 0.07, 0.10),
+                           proportions = c(0.079, 0.134, 0.139, 0.082, 0.066),
+                           counts = NULL,
                            tibble = TRUE) {
 
-  if (length(proportions) != length(groups) & is.null(strata)) {
-    difference <- abs(length(groups) - length(proportions))
+  # pick counts if not empty
+  measure <- if (is.null(counts)) proportions else counts
 
-    warning(sprintf("Proporitons and groups do not match, without specifying strata.
-                    The difference in length was %d (proportions may have been dropped)",
-                    difference))
-
-    proportions <- proportions[1:length(groups)]
-  }
+  # define lengths of inputs for checks
+  ngroups <- length(groups)
+  nstrata <- length(strata)
+  nmeasure <- length(measure)
 
+  # define a dataframe based on groups only
+  output <- data.frame(groups)
```

Comment:
Move this to the "else" block below since it's the alternative.

Reply:
so its actually used in the "else" block but then also in the "if (is.null(counts))" section ... thats why i left it here

Reply:
oh but i can move it to else anyway sorry you are right

Reply:
but the `output` variable is not used anywhere between lines 90 and 99 (the else block). What you are effectively doing is this (hiding the cromulence checks):

```r
output <- data.frame(groups)

if (!is.null(strata)) {
  # overwrite output variable that we created above
  output <- expand.grid(groups, strata)
} else {
  # ¯\_(ツ)_/¯
}
```

Instead, it would be better to do this:

```r
if (!is.null(strata)) {
  output <- expand.grid(groups, strata)
} else {
  output <- data.frame(groups)
}
```